test/runtests.jl

```diff
@@ -78,3 +78,22 @@ include("objgrad-on-tron.jl")
   nls = ADNLSModel(x -> [100 * (x[2] - x[1]^2); x[1] - 1], [-1.2; 1.0], 2)
   stats = tron(nls, max_radius = max_radius, increase_factor = increase_factor, callback = cb)
 end
+
+@testset "Preconditioner in Trunk" begin
+  x0 = [-1.2; 1.0]
+  nlp = ADNLPModel(x -> 100 * (x[2] - x[1]^2)^2 + (x[1] - 1)^2, x0)
+  function DiagPrecon(x)
+     H = Matrix(hess(nlp, x))
+     λmin = minimum(eigvals(H))
+     Diagonal(H + λmin * I)
```

Comment:
Just so users aren't confused, we should add a little more than λmin here. If H has a negative eigenvalue, this preconditioner is singular.